src/api/Item.js

```diff
@@ -302,37 +327,66 @@ class Item extends Base {
             return Promise.reject();
         }
 
-        this.errorCode = ERROR_CODE_SHARE_ITEM;
-        const { id, permissions }: BoxItem = item;
-        if (!id || !permissions) {
-            errorCallback(getBadItemError(), this.errorCode);
-            return Promise.reject();
-        }
+        try {
+            const { id, permissions }: BoxItem = item;
+            this.id = id;
+            this.successCallback = successCallback;
+            this.errorCallback = errorCallback;
 
-        const { can_share, can_set_share_access }: BoxItemPermission = permissions;
-        if (!can_share || !can_set_share_access) {
-            errorCallback(getBadPermissionsError(), this.errorCode);
-            return Promise.reject();
-        }
+            this.validateRequest(id, permissions);
 
-        const cache: APICache = this.getCache();
-        const key: string = this.getCacheKey(id);
-        const isCached: boolean = !options.forceFetch && cache.has(key);
+            const { fields } = options;
+            const requestData: RequestData = {
+                url: this.getUrl(this.id),
+                data: {
+                    shared_link: access === ACCESS_NONE ? null : { access },
+                },
+            };
+            if (fields) {
+                requestData.params = { fields: fields.toString() };
+            }
```

Comment:
Maybe can use `Object.assign` to get rid of `if`, like `Object.assign(requestData, fields && { params: {fields: fields.toString()}})`